backend/fms_core/services/study.py

```diff
@@ -1,8 +1,29 @@
 from django.core.exceptions import ValidationError
-from fms_core.models import Study, Project
+from fms_core.models import Study, Project, StepHistory, SampleNextStep, SampleNextStepByStudy
 
+from collections import defaultdict
+from typing import Dict, List, Tuple
 import string
 
+def new_letter(project):
+    def skip_char(letter: str, distance: int):
+        return chr(ord(letter) + distance)
+
+    FIRST_LETTER = 'A'
+    occupied_letters = sorted(Study.objects.filter(project=project).values_list('letter', flat=True))
+    new_letter = skip_char(max(occupied_letters), 1) if len(occupied_letters) > 0 else FIRST_LETTER
```

Comment:
if you sort your list (as you do here), you should slice here instead of using max. The last item will be your max,